src/index.ts

```diff
@@ -1,2 +1,143 @@
+            throw new Error("Title and description cannot be empty.");
+        }
+        this.id = TodoItem.currentId++;
+        this.title = title;
+        this.description = description;
+        this.creationDate = new Date();
+        this.lastEditDate = new Date();
+        this.isCompleted = false;
+        this.isEditable = isEditable;
+        this.type = type;
+    }
+
+    // Метод для позначення нотатки як виконаної
+    markAsCompleted(): void {
+        this.isCompleted = true;
+    }
+
+    // Метод для оновлення дати останнього редагування
+    updateLastEditDate(): void {
+        this.lastEditDate = new Date();
+    }
+}
+
+class TodoList {
+    private items: TodoItem[] = [];
+
+    addItem(title: string, description: string, type: NoteType = NoteType.DEFAULT): TodoItem {
+        const newItem = new TodoItem(title, description, type);
+        this.items.push(newItem);
+        return newItem;
+    }
+
+    // Метод для видалення запису за ідентифікатором
+    removeItem(id: number): boolean {
+        const index = this.items.findIndex(item => item.id === id);
+        if (index !== -1) {
+            this.items.splice(index, 1);
+            return true;
+        }
+        return false;
+    }
+
+    // Метод для редагування запису за ідентифікатором
+    editItem(id: number, title: string, description: string, type: NoteType): boolean {
+        const item = this.items.find(item => item.id === id);
+        if (item) {
+            if (item.isEditable) {
+                item.title = title;
+                item.description = description;
+                item.type = type; // Оновлення типу нотатки
+                item.updateLastEditDate();
+                return true;
+            } else {
+                item.title = title;
+                item.description = description;
+                item.type = type; // Оновлення типу нотатки
+                return true;
+            }
+        }
+        return false;
+    }
+
+    // Метод для отримання повної інформації про нотатку за ідентифікатором
+    getItemById(id: number): TodoItem | undefined {
+        return this.items.find(item => item.id === id);
+    }
+
+    // Метод для отримання списку всіх нотаток
+    getAllItems(): TodoItem[] {
+        return this.items;
+    }
+
+    // Метод для позначення нотатки як виконаної
+    markItemAsCompleted(id: number): boolean {
+        const item = this.getItemById(id);
+        if (item) {
+            item.markAsCompleted();
+            return true;
+        }
+        return false;
+    }
+
+    // Метод для пошуку нотатки за ім'ям або змістом
+    searchItem(query: string): TodoItem[] {
+        query = query.toLowerCase();
+        return this.items.filter(item => item.title.toLowerCase().includes(query) || item.description.toLowerCase().includes(query));
+    }
+
+    // Метод для сортування нотаток за статусом (виконано / невиконано)
+    sortItemsByStatus(): void {
```

Comment:
This method should be implemented separately from the base class, creating another entity.